pubsub/google/cloud/pubsub_v1/gapic/publisher_client.py

```diff
@@ -114,14 +114,24 @@ def from_service_account_file(cls, filename, *args, **kwargs):
 
     @classmethod
     def project_path(cls, project):
-        """Return a fully-qualified project string."""
+        """DEPRECATED. Return a fully-qualified project string."""
+        warnings.warn(
+            "Resource name helper functions are deprecated.",
+            PendingDeprecationWarning,
```

Comment:
If these methods are already deprecated now, wouldn't it be more correct to issue a `DeprecationWarning` instead? Or is this a conscious choice?

Asking because the [docs](https://docs.python.org/3/library/exceptions.html#warnings) recommend raising that warning type as _"emitting a warning about a possible upcoming deprecation is unusual"_.

Reply:
Ah that makes sense. I think we went with `PendingDeprecationWarning` because it's already used in [other parts of the code](https://github.com/googleapis/google-cloud-python/search?q=PendingDeprecationWarning&unscoped_q=PendingDeprecationWarning).

Reply:
Yeah, saw plenty of those across the repository ... enough to become a bit suspicious. :)

Reply:
Sorry for being out of the loop: Why are these being deprecated?

Reply:
These methods cannot be supported in the microgenerator (the new Python generator we'll start using Q1/Q2 of next year), so we are deprecating them. See https://github.com/googleapis/gapic-generator/pull/3016

Reply:
Sounds good.